tao_compiler/mlir/disc/transforms/fusion_utils.h

```diff
@@ -465,6 +472,46 @@ class FusionStrategy {
   FusionOptions options_;
 };
 
+using DeviceStrategyMap = DenseMap<StringRef, FusionStrategy*>;
+
+class PlacementAwareFusionStrategy : public FusionStrategy {
+ public:
+  PlacementAwareFusionStrategy(const FusionOptions& options,
+                               StringRef defaultDevice,
+                               DeviceStrategyMap deviceStrategyMap)
+      : FusionStrategy(options),
+        deviceStrategyMap_(std::move(deviceStrategyMap)),
+        defaultDevice_(defaultDevice) {}
+
+  bool isFusible(Operation* op) override;
+  bool isFusible(FusionPattern& fusion_pattern) override;
+  bool tryFuse(ShapeAnalysis& shapeAnalysis, FusionPattern& lhs,
+               FusionPattern& rhs, FusionPattern& target) override;
+  bool initFusionPattern(ShapeAnalysis& shapeAnalysis,
+                         FusionPattern& fusion_pattern) override;
+  virtual bool finalizeFusionPattern(
```

Comment:
What's the supposed semantic of such function?

Reply:
Already discussed offline.